cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -0,0 +1,283 @@
+        ret_val, json_out = self._common_cmd_json_gen(cmd_json)
+        # if there is an unsuccessful attempt,
+        if ret_val != 0:
+            raise ExecutionFailureException(
+                "'auth get-or-create client.csi-rbd-node' command failed")
+        if len(json_out) > 0:
+            return str(json_out[0]['key'])
+        raise ExecutionFailureException(
+            "Failed to get/create 'client.csi-rbd-node' key")
+
+    def create_checkerKey(self):
+        cmd_json = {"prefix": "auth get-or-create",
+                    "entity": self.run_as_user,
+                    "caps": ["mon", "allow r, allow command quorum_status"],
+                    "format": "json"}
+        ret_val, json_out = self._common_cmd_json_gen(cmd_json)
+        # if there is an unsuccessful attempt,
+        if ret_val != 0:
+            raise ExecutionFailureException(
+                "'auth get-or-create {}' command failed".format(self.run_as_user))
+        if len(json_out) > 0:
+            return str(json_out[0]['key'])
+        raise ExecutionFailureException(
+            "Failed to get/create 'client.csi-rbd-node' key")
+
+    def _gen_output_map(self):
+        if self.out_map:
+            return
+        self.out_map['NAMESPACE'] = self._arg_parser.namespace
+        self.out_map['ROOK_EXTERNAL_FSID'] = self.get_fsid()
+        self.out_map['ROOK_EXTERNAL_USERNAME'] = self._arg_parser.run_as_user
+        self.out_map['ROOK_EXTERNAL_CEPH_MON_DATA'] = self.get_ceph_external_mon_data()
+        self.out_map['ROOK_EXTERNAL_USER_SECRET'] = self.create_checkerKey()
+        self.out_map['CSI_RBD_NODE_SECRET_SECRET'] = self.create_cephCSIKeyring_RBDNode()
+        self.out_map['CSI_RBD_PROVISIONER_SECRET'] = self.create_cephCSIKeyring_RBDProvisioner()
+        self.out_map['CSI_CEPHFS_NODE_SECRET'] = self.create_cephCSIKeyring_cephFSNode()
+        self.out_map['CSI_CEPHFS_PROVISIONER_SECRET'] = self.create_cephCSIKeyring_cephFSProvisioner()
+        self.out_map['RGW_GATEWAY_ENDPOINT'] = self._arg_parser.rgw_endpoint
+        self.out_map['CEPHFS_POOL_NAME'] = self._arg_parser.cephfs_data_pool_name
+        self.out_map['CEPHFS_FS_NAME'] = self._arg_parser.cephfs_filesystem_name
+        self.out_map['RBD_POOL_NAME'] = self._arg_parser.rdb_datapool_name
+        self.out_map['RGW_POOL_PREFIX'] = self._arg_parser.rgw_pool_prefix
+        self.out_map['RGW_REALM_NAME'] = self._arg_parser.rgw_realm_name
+        self.out_map['RGW_ZONEGROUP_NAME'] = self._arg_parser.rgw_zonegroup_name
+        self.out_map['RGW_ZONE_NAME'] = self._arg_parser.rgw_zone_name
+
+    def gen_shell_out(self):
+        self._gen_output_map()
+        shOutIO = StringIO()
+        for k, v in self.out_map.items():
+            if v:
+                shOutIO.write('export {}={}{}'.format(k, v, LINESEP))
+        shOut = shOutIO.getvalue()
+        shOutIO.close()
+        return shOut
+
+    def gen_json_out(self):
+        self._gen_output_map()
+        json_out = [
```

Comment:
We need all the CSI secret and the checker key too.

Reply:
@leseb we need the cephcsi configmap, who generates that? rook or that also needs to be part of this file

Reply:
Rook creates the CSI configmap automatically.